app/javascript/components/modals/Modal.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { useLayoutEffect } from 'react'
+import { default as ReactModal } from 'react-modal'
+
+export const Modal = ({
+  open,
+  onClose,
+  children,
+  ...props
+}: {
+  open: boolean
+  onClose: () => void
+  children?: React.ReactNode
+}): JSX.Element => {
+  useLayoutEffect(() => {
+    // Required for accessibility: http://reactcommunity.org/react-modal/accessibility/#app-element
+    ReactModal.setAppElement(document.querySelector('body') as HTMLElement)
```

Comment:
This is the main reason why I introduced this component, as we need to set the app element once the DOM is ready.